atomic_reactor/source.py

```diff
@@ -65,6 +72,25 @@ def __init__(self, build_path):
         self.remote_sources = self.data.get('remote_sources')
         self.operator_manifests = self.data.get('operator_manifests')
 
+        # [] is the internal representation of not or only is not set in config.
+        # "not set" means:
+        #   1) none of them is present
+        #   2) any of them is present but without values
+        #
+        self.platforms = self.data.get('platforms') or {'not': [], 'only': []}
+        self.platforms['not'] = make_list(self.platforms.get('not', []))
+        self.platforms['only'] = make_list(self.platforms.get('only', []))
+        if self.platforms['not'] == self.platforms['only']:
+            logger.warning('only and not platforms are the same in %s', self.file_path)
```

Comment:
I even think we don't need this log. As long as the code is correct to handle platforms, whether the "not" and "only" `platforms` equal or not, in most case, it is not an important thing we have to pay attention to. On the other hand, if we log these platforms, logging either of them in debug level is enough, like:
```python
logger.debug("only build for platforms: %s", self.platforms["only"])
logger.debug("excludes platforms: %s", self.platforms["not"])
```
There are only a few platforms. It's easy for log readers to know whether they are same or not. We don't need code to tell that.

Reply:
I actually wonder if this should have been just warning ...

I'd actually fail if you specify ANY platform in both not & only, as that is basically already error, our code is processing it anyway as it first uses only & then excludes, but this is basically hiding user errors

Reply:
> I'd actually fail if you specify ANY platform in both not & only, as that is basically already error, our code is processing it anyway as it first uses only & then excludes, but this is basically hiding user errors

It seems fine to me, people may have their reasons for using it that way. Besides, it would be a breaking change.

Reply:
well there isn't any reason why they should have same arch specified in both not & only

Reply:
Yeah, I'm really having trouble imagining a scenario which makes any sense at all where someone is saying "Give me ONLY s390x but also COMPLETELY EXCLUDE s390x" 🤨 - seems like they **should** be mutually exclusive

Reply:
I could imagine someone doing this
```yaml
platforms:
  # these are the platforms we support long term
  only: ["x86_64", "aarch64"]
  # temporarily exclude arm because we are having trouble with it
  not: "aarch64"
```
Not saying it's a particularly important use case or even that anyone actually uses it. Just trying to illustrate that there could be a use case and we should not break it.